Libplanet/Net/BlockCompletion.cs

```diff
@@ -328,10 +251,98 @@ await foreach (Block<TAction> block in blocks)
                     pair.Item1.Hash,
                     pair.Item2
                 );
-                completion[pair.Item1.Hash] = true;
             }
 
-            _logger.Verbose("Completed all blocks ({Number}).", completion.Count);
+            Func<TPeer, CancellationToken, Task> CreateBatch(
```

Comment:
How about extract this to a separated `private` method taking `blockFetcher`?

Reply:
That's the possible option. but it can be quite messy due to its arguments...

```csharp
private Func<TPeer, CancellationToken, Task> CreateBatch(
            IList<HashDigest<SHA256>> hashDigests,
            BlockFetcher blockFetcher,
            TimeSpan singleSessionTimeout,
            CancellationToken cancellationToken,
            AsyncProducerConsumerQueue<Tuple<Block<TAction>, TPeer>> queue
        ) =>
```